File: Frontend/src/app/shared/models/booking.ts
export interface Booking {
  bookingId: number,
  venueName: string,
  location: string,
  startDateTime: string,
  endDateTime: string,
  totalPrice: number,
  bookingStatus?: string,
  imagePath?:string
}

export const RazorPayment = {
  razorpayKey: 'rzp_test_T71Ifrp2xFrEPh',
};

export interface BookingRequestDTO {
  venueId: number;
  slotId: number;
  startDateTime: string; // ISO date string
  endDateTime: string;
}

export interface OrderRequestDTO {
  bookingId: number;
}

File: backend/src/main/java/com/bookmyvenue/backend/entity/Booking.java
package com.bookmyvenue.backend.entity;

import com.bookmyvenue.backend.enums.BookingStatus;
import com.bookmyvenue.backend.enums.PaymentStatus;
import com.bookmyvenue.backend.enums.SlotType;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long bookingId;

    @ManyToOne
    @JoinColumn(name = "slot_id")
    private Slot slot;

    @ManyToOne
    @JoinColumn(name = "venue_id")
    private Venue venue;

    @ManyToOne
    @JoinColumn(name = "userId")
    private User user;

    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
    private SlotType slotType;
    private BigDecimal totalPrice;

    @Enumerated(EnumType.STRING)
    private BookingStatus bookingStatus;

    private String razorpayOrderId;

    private String razorpayPaymentId;

    @Enumerated(EnumType.STRING)
    private PaymentStatus paymentStatus;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}